cmd/janitor/gcp_janitor.py

```diff
@@ -508,17 +508,7 @@ def activate_service_account(service_account):
 
 def set_quota_project(project):
     print('[=== Setting quota_project %s ===]' % project)
-    cmd = [
-        'gcloud', 'config', 'set', 'billing/quota_project',
-        '%s' % project,
-    ]
-    log('running %s' % cmd)
-
-    try:
-        subprocess.check_call(cmd)
-    except subprocess.CalledProcessError:
-        print('Error try to set quota_project: %s' % project, file=sys.stderr)
-        return 1
+    os.environ['CLOUDSDK_BILLING_QUOTA_PROJECT'] = '%s' % project
```

Comment:
My Python is rusty so I stackoverflowed and I found https://stackoverflow.com/questions/5971312/how-to-set-environment-variables-in-python#comment61988654_5971326 --- do the concerns there about child processes vs parent process apply here as well? I assume "no" and this is fine, but just wanted to ask.

Reply:
Hey Linus: thanks for the prompt attention. To ensure I'm responding to your actual question, you're referring to this comment, yes?

> Caution: to quote from @Evan's reference above, Such changes to the environment affect subprocesses started with os.system(), popen() or fork() and execv(). In other words, keep in mind that this approach won't modify the way your program is running, only the way your program's children run.

In this case, `gcp_janitor.py` *is* the child process (it's spawned from `janitor.go`). So, settings that we make in `gcp_janitor.py` only affect its own specific invocation, not others spawned by `janitor.go`.

You can see that overriding the environment variable *does* affect subsequent `subprocess` invocations in the same script. Some sample code:
```python
import os, subprocess

GCLOUD_CMD = "gcloud config get api_endpoint_overrides/gkehub"

def set_gkehub(api):
  os.environ["CLOUDSDK_API_ENDPOINT_OVERRIDES_GKEHUB"] = f'https://{api}/'

def run_cmd(pstr):
  print(f"-- {pstr}")
  subprocess.run(GCLOUD_CMD, shell=True, text=True, check=True)

run_cmd("No env override")

gkehub_apis = {'gkehub.googleapis.com', 'staging-gkehub.sandbox.googleapis.com', 'autopush-gkehub.sandbox.googleapis.com'}
for api in gkehub_apis:
  set_gkehub(api)
  run_cmd(f"Env override with {api}")
```
And here's the output when I run that code:
```
-- No env override
(unset)
-- Env override with autopush-gkehub.sandbox.googleapis.com
https://autopush-gkehub.sandbox.googleapis.com/
-- Env override with staging-gkehub.sandbox.googleapis.com
https://staging-gkehub.sandbox.googleapis.com/
-- Env override with gkehub.googleapis.com
https://gkehub.googleapis.com/
```
The bottom line is: `os.environ` sets the environment only for a given script and its children, and separate invocations of `gce_janitor.py` (which don't affect each others' environments) are spawned by `janitor.go`.

Reply:
Sorry, Linus, I should add one clarification:

Calls to `subprocess` use `popen` under the hood. That's why setting `os.environ` inside my sample code above does affect subsequent calls to `subprocess`.

Reply:
> To ensure I'm responding to your actual question, you're referring to this comment, yes?

Yes. (Sorry I didn't make it clear in my original question.)

LGTM

Reply:
Thank you, @listx!

By the way, I see a label on this PR saying `needs-ok-to-test`. Are you able to authorize that?